corporate/lib/stripe.py

```diff
@@ -2347,6 +2347,7 @@ def switch_plan_tier(self, current_plan: CustomerPlan, new_plan_tier: int) -> No
             licenses_at_next_renewal=licenses_for_new_plan,
         )
 
+    @transaction.atomic
```

Comment:
Hmm, this change might be dangerous. The problem is that the Stripe operations are side effects that don't happen inside the Zulip database, and so having this whole thing be a transaction means that it's possible for changes to happen in Stripe without appearing in Zulip.

I don't really know what the best answer for this is -- but usually it involves multiple commit phases.